vendor/github.com/docker/distribution/registry/client/blob_writer.go

```diff
@@ -99,7 +99,7 @@ func (hbu *httpBlobUpload) Write(p []byte) (n int, err error) {
 		return 0, fmt.Errorf("bad range format: %s", rng)
 	}
 
-	return (end - start + 1), nil
```

Comment:
Yes, it has, in general this fix is right and should be upstreamed. But I'm more curious how do you find this? A sequence of writes will produce a sequence of HTTP requests while ReadFrom produces only one HTTP request. And the sequence of HTTP requests not only utilises network less efficient, but may fail on the registry side if it hits S3 (temporary) read-after-update inconsistency. So technically it implements io.Write and it works, but it's better to pretend it's just io.ReaderFrom. Fortunately io.Copy helps with it and uses ReadFrom if it can.